api/tasks.py

```diff
@@ -24,21 +24,24 @@
 @dramatiq.actor(actor_name="poll_updates", max_retries=MAX_RETRIES)
 @settings.run_sync
 async def poll_updates(obj: Union[int, models.Invoice], task_wallets: Dict[str, str]):
-    test = settings.TEST
     obj = await models.Invoice.get(obj)
     if not obj:
         return
     await crud.invoice_add_related(obj)
+    if settings.TEST:
+        await asyncio.sleep(1)
+        await obj.update(status="test").apply()
+        await utils.publish_message(obj.id, {"status": "test"})
+        return
     payment_methods = await models.PaymentMethod.query.where(models.PaymentMethod.invoice_id == obj.id).gino.all()
     if not payment_methods:
         return
     for ind, method in enumerate(payment_methods):
         payment_methods[ind].coin = settings.get_coin(method.currency, task_wallets[method.currency])
-    if test:
-        await asyncio.sleep(1)
-        await obj.update(status="test").apply()
-        await utils.publish_message(obj.id, {"status": "test"})
-        return
+    return await process_invoice(obj, task_wallets, payment_methods)
```

Comment:
Dramatiq will require installing Results middleware to process results. `process_invoice` doesn't return anything too, but anyway
```suggestion
    await process_invoice(obj, task_wallets, payment_methods)
```